coriolis/osmorphing/osmount/base.py

```diff
@@ -258,6 +249,28 @@ def _check_mount_fstab_partitions(
 
         return new_mountpoints
 
+    def _get_symlink_target(self, symlink):
+        target = ''
+        try:
+            target = self._exec_cmd('readlink -en %s' % symlink).decode()
+        except Exception:
+            LOG.warn('Target not found for symlink: %s' % symlink)
+
+        return target
+
+    def _get_device_file_paths(self, symlink_list):
+        """ Reads a list of symlink paths, such as `/dev/GROUP/VOLUME0` or
+        `/dev/mapper/GROUP-VOLUME0` and returns a list of respective links'
+        target device file paths, such as `/dev/dm0`
+        """
+        dev_file_paths = []
+        for link in symlink_list:
+            dev_file = self._get_symlink_target(link)
+            if not dev_file:
```

Comment:
This means that if for whatever reason there is a file which isn't a symlink it'll get completely "lost in translation"
IMO either this method should be re-written to just return the same path if something is not a symlink